treeherder/webapp/api/performance_data.py

```diff
@@ -751,6 +752,31 @@ def _filter_out_retriggers(serialized_data: List[dict]) -> List[dict]:
         return serialized_data
 
 
+class PerformanceAlertSummaryTasks(generics.ListAPIView):
+    serializer_class = PerformanceAlertSummaryTasksSerializer
+    queryset = None
+
+    def list(self, request):
+        query_params = self.get_serializer(data=request.query_params)
```

Comment:
I believe this serializer is a query param serializer, so instead it should be declared like so:
```suggestion
        query_params = PerfSummaryTasksQueryParamSerializer(data=request.query_params)
```
This means you would need another serializer for the `serializer_class` field. 

Another approach to validate the id is to move the logic from `validate` directly into the `list` method using `request.query_params.get('id')` (there are examples for this in the code base).

Reply:
The reason behind my comment is that I haven't seen the same serializer used for both fields `serializer_class` and `query_params` before. I am definitely not an expert so your approach might be fine, but I wanted to share my thoughts.

@gmierz ^

Reply:
If there are some kinds of `Query Param`, I thought it would be good to make Serializer for the query param. But I didn't make it separately because there is only one that is `id`.
If it's a good way to separate, I'll make a new serializer for query parameters :)

Reply:
Good catch @beatrice-acasandrei! I think we should follow what the other methods are doing, and add a second serializer for the query params like in `PerformanceSummary`: https://github.com/mozilla/treeherder/pull/7717/files#diff-c1e40d4858d9a69a025cfa032d9681b896632ca12bda3819c9833d0589776254R613-R615